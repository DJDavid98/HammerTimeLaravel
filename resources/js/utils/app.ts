import { rangeLimitInput } from '@/utils/time';
import { Ref } from 'vue';

export const getAppName = () => import.meta.env.VITE_APP_NAME || 'Laravel';

export const inputRangeLimitBlurHandlerFactory = (numberRef: Ref<number>) => (e: FocusEvent): void => {
  const limitedValue = rangeLimitInput(e.target);
  if (limitedValue === numberRef.value)
    return;

  numberRef.value = limitedValue;
};
