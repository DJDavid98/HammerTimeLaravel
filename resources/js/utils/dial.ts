export enum DialMode {
  Hours = 'hours',
  Minutes = 'minutes',
  Seconds = 'seconds'
}
