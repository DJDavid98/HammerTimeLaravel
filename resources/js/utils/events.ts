export const keyboardOrMouseEventHandlerFactory = (handler: (e: KeyboardEvent | MouseEvent, viaKeyboard: boolean) => void) =>
  (e: KeyboardEvent | MouseEvent) => {
    let viaKeyboard = false;
    if ('key' in e) {
      if (e.key === 'Tab' || e.key === 'Shift') {
        return;
      }
      viaKeyboard = true;
    }

    e.preventDefault();
    handler(e, viaKeyboard);
  };
