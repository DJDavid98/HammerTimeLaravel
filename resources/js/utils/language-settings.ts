import { LanguageConfig } from '@/model/language-config';
import localeConfig from '../../../lang/config.json';

export type AvailableLanguage = keyof typeof localeConfig.languages;

type LanguagesConfig = Record<AvailableLanguage, LanguageConfig>;

export const LANGUAGES = localeConfig.languages as LanguagesConfig;

export const isAvailableLanguage = (language: string): language is AvailableLanguage => language in LANGUAGES;

