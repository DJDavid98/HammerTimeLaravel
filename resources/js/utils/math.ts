export interface Point2D {
  x: number;
  y: number;
}

export type RequiredDomRectInit = Required<DOMRectInit>;

enum PointQuadrant {
  PosXNegY,
  PosXPosY,
  NegXNegY,
  NegXPosY,
}

const translateOrigin = (originalPoint: Point2D, newOrigin: Point2D, previousOrigin: Point2D = {
  x: 0,
  y: 0,
}) => {
  const offsetX = newOrigin.x - previousOrigin.x;
  const offsetY = newOrigin.y - previousOrigin.y;

  return { x: originalPoint.x - offsetX, y: originalPoint.y - offsetY };
};

export const getPointQuadrant = (point: Point2D): PointQuadrant => {
  if (point.x > 0) {
    if (point.y > 0) {
      return PointQuadrant.PosXPosY;
    }

    return PointQuadrant.PosXNegY;
  }

  if (point.y > 0) {
    return PointQuadrant.NegXPosY;
  }

  return PointQuadrant.NegXNegY;
};

const radiansToDegrees = (radians: number) => radians * (180 / Math.PI);

const calculateAngle = (a: Point2D, b: Point2D, c: Point2D): number => {
  const ab = Math.sqrt(Math.pow(b.x - a.x, 2) + Math.pow(b.y - a.y, 2));
  const bc = Math.sqrt(Math.pow(b.x - c.x, 2) + Math.pow(b.y - c.y, 2));
  const ac = Math.sqrt(Math.pow(c.x - a.x, 2) + Math.pow(c.y - a.y, 2));
  const angleInRadians = Math.acos((Math.pow(ab, 2) + Math.pow(bc, 2) - Math.pow(ac, 2)) / (2 * ab * bc));
  return radiansToDegrees(angleInRadians);
};

/**
 * Calculate the angle the position is in relative to a circle at origin (0,0),
 * where 0 degrees is pointing north
 *
 * @return angle (0-359)
 */
export const getPositionAngle = (point: Point2D): number => {
  const pointQuadrant = getPointQuadrant(point);
  switch (pointQuadrant) {
    case PointQuadrant.NegXPosY:
      if (point.x === 0) {
        return 180;
      }
      break;
    case PointQuadrant.NegXNegY:
      if (point.x === 0) {
        return 0;
      }
      if (point.y === 0) {
        return 270;
      }
      break;
    case PointQuadrant.PosXNegY:
      if (point.y === 0) {
        return 90;
      }
      break;
  }

  const otherVertex: Point2D = {
    x: point.x,
    y: 0,
  };
  const rightAngleVertex: Point2D = {
    x: 0,
    y: point.y,
  };
  const angle = calculateAngle(point, rightAngleVertex, otherVertex);
  switch (pointQuadrant) {
    case PointQuadrant.PosXNegY:
      return 90 - angle;
    case PointQuadrant.PosXPosY:
      return angle + 90;
    case PointQuadrant.NegXPosY:
      return 90 - angle + 180;
    case PointQuadrant.NegXNegY:
      return angle + 270;
  }
};

export const calculateFractionalDistance = (point: Point2D, maxWidth: number = 1): number => {
  // Calculate the distance from the origin to the point
  const distance = Math.sqrt(Math.pow(point.x, 2) + Math.pow(point.y, 2));

  return distance / maxWidth;
};

export interface PositionData {
  angle: number;
  distancePercent: number;
}

/**
 * Given the bounding box & position of an element on the page and
 * a specific point on the page, calculate the angle the position
 * is in relative to the circle with a diameter and origin matching
 * the element
 *
 * @return angle (0-359)
 */
export const getPositionAngleInElement = (elementRect: RequiredDomRectInit, originalPoint: Point2D): PositionData => {
  const elementWidthHalf = elementRect.width / 2;
  const elementHeightHalf = elementRect.height / 2;
  const elementCenterPosition: Point2D = {
    x: Math.round(elementRect.x + elementWidthHalf),
    y: Math.round(elementRect.y + elementHeightHalf),
  };
  const point = translateOrigin(originalPoint, elementCenterPosition);
  const maxDistance = calculateFractionalDistance({
    x: elementWidthHalf,
    y: elementHeightHalf,
  }, 1);
  return {
    angle: getPositionAngle(point),
    distancePercent: calculateFractionalDistance(point, maxDistance),
  };
};

export const integerInRangeByAngle = (minValue: number, maxValue: number, angle: number) => {
  const value = Math.round(minValue + (angle / 360) * (maxValue - minValue));
  // If value exceeds maximum, wrap around intentionally
  return value > maxValue - 1 ? minValue : value;
};
