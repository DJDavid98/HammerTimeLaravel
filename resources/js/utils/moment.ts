import moment from 'moment/moment';
import { momentLocaleMap } from '@/moment';

export const useMomentLocale = async (locale: string): Promise<void> => {
  if (locale in momentLocaleMap) {
    await momentLocaleMap[locale as keyof typeof momentLocaleMap]();
    moment.locale(locale);
  }
};
