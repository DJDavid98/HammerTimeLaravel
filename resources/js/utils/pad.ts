export const pad = (value: number | string, digits: number) => {
  const valueString = typeof value === 'number' ? value.toString() : value;
  if (valueString.length < digits) {
    return '0'.repeat(digits - valueString.length) + valueString;
  }
  return valueString;
};
