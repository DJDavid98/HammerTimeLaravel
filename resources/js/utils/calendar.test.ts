import {
  DayOfWeek,
  generateCalendar,
  getFirstDayOfWeekOffset,
  getMomentForDate,
  Month,
} from '@/utils/calendar';
import { isoParsingDateFormat } from '@/utils/timezone';
import { describe, expect, it } from 'vitest';

describe('Calendar Utils', () => {
  const year = 2023;
  const month = Month.May;

  describe('getMomentForDate', () => {
    it('should get the correct moment for a date', () => {
      expect(getMomentForDate(2022, Month.December, 1).format(isoParsingDateFormat)).to.eql('2022-12-01');
      expect(getMomentForDate(2023, Month.January, 1).format(isoParsingDateFormat)).to.eql('2023-01-01');
      expect(getMomentForDate(2023, Month.February, 1).format(isoParsingDateFormat)).to.eql('2023-02-01');
      expect(getMomentForDate(2023, Month.March, 1).format(isoParsingDateFormat)).to.eql('2023-03-01');
      expect(getMomentForDate(2023, Month.April, 1).format(isoParsingDateFormat)).to.eql('2023-04-01');
      expect(getMomentForDate(2023, Month.May, 1).format(isoParsingDateFormat)).to.eql('2023-05-01');
      expect(getMomentForDate(2023, Month.June, 1).format(isoParsingDateFormat)).to.eql('2023-06-01');
      expect(getMomentForDate(2023, Month.July, 1).format(isoParsingDateFormat)).to.eql('2023-07-01');
      expect(getMomentForDate(2023, Month.August, 1).format(isoParsingDateFormat)).to.eql('2023-08-01');
      expect(getMomentForDate(2023, Month.September, 1).format(isoParsingDateFormat)).to.eql('2023-09-01');
      expect(getMomentForDate(2023, Month.October, 1).format(isoParsingDateFormat)).to.eql('2023-10-01');
      expect(getMomentForDate(2023, Month.November, 1).format(isoParsingDateFormat)).to.eql('2023-11-01');
      expect(getMomentForDate(2023, Month.December, 1).format(isoParsingDateFormat)).to.eql('2023-12-01');
    });
  });

  describe('getFirstDayOfWeekOffset', () => {
    it('should return the correct offset for a month that starts on Monday and the first day of week is Monday', () => {
      expect(getFirstDayOfWeekOffset(getMomentForDate(year, month, 1), DayOfWeek.Monday)).to.eql(0);
    });
    it('should return the correct offset for a month that starts on Monday and the first day of week is Tuesday', () => {
      expect(getFirstDayOfWeekOffset(getMomentForDate(year, month, 1), DayOfWeek.Tuesday)).to.eql(-6);
    });
    it('should return the correct offset for a month that starts on Monday and the first day of week is Wednesday', () => {
      expect(getFirstDayOfWeekOffset(getMomentForDate(year, month, 1), DayOfWeek.Wednesday)).to.eql(-5);
    });
    it('should return the correct offset for a month that starts on Monday and the first day of week is Thursday', () => {
      expect(getFirstDayOfWeekOffset(getMomentForDate(year, month, 1), DayOfWeek.Thursday)).to.eql(-4);
    });
    it('should return the correct offset for a month that starts on Monday and the first day of week is Friday', () => {
      expect(getFirstDayOfWeekOffset(getMomentForDate(year, month, 1), DayOfWeek.Friday)).to.eql(-3);
    });
    it('should return the correct offset for a month that starts on Monday and the first day of week is Saturday', () => {
      expect(getFirstDayOfWeekOffset(getMomentForDate(year, month, 1), DayOfWeek.Saturday)).to.eql(-2);
    });
    it('should return the correct offset for a month that starts on Monday and the first day of week is Sunday', () => {
      expect(getFirstDayOfWeekOffset(getMomentForDate(year, month, 1), DayOfWeek.Sunday)).to.eql(-1);
    });
    it('should return the correct offset for a month that starts on Friday and the first day of week is Monday', () => {
      expect(getFirstDayOfWeekOffset(getMomentForDate(year, Month.December, 1), DayOfWeek.Monday)).to.eql(-4);
    });
  });

  describe('generateCalendar', () => {
    it('should generate the correct calendar for a month that starts on Monday and the first day of week is Monday', () => {
      const actual = generateCalendar({
        year, month, firstDayOfWeek: DayOfWeek.Monday,
      });
      expect(actual).to.matchSnapshot();
    });

    it('should generate the correct calendar for a month that starts on Friday and the first day of week is Monday', () => {
      const actual = generateCalendar({
        year: 2023, month: Month.December, firstDayOfWeek: DayOfWeek.Monday,
      });
      expect(actual).to.matchSnapshot();
    });


    it('should generate the correct calendar for a month that starts on Thursday and the first day of week is Monday', () => {
      const actual = generateCalendar({
        year: 2022, month: Month.December, firstDayOfWeek: DayOfWeek.Monday,
      });
      expect(actual).to.matchSnapshot();
    });

    it('should generate the correct calendar for a month that starts on Monday and the first day of week is Sunday', () => {
      const actual = generateCalendar({
        year, month, firstDayOfWeek: DayOfWeek.Sunday,
      });
      expect(actual).to.matchSnapshot();
    });

    it('should generate the correct calendar for a month that starts on Monday and the first day of week is Saturday', () => {
      const actual = generateCalendar({
        year, month, firstDayOfWeek: DayOfWeek.Saturday,
      });
      expect(actual).to.matchSnapshot();
    });

    it('should generate the correct calendar for a month that starts on Saturday and the first day of week is Sunday', () => {
      const actual = generateCalendar({
        year: 2023, month: Month.April, firstDayOfWeek: DayOfWeek.Sunday,
      });
      expect(actual).to.matchSnapshot();
    });
  });
});
