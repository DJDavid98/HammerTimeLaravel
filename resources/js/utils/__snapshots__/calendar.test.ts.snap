// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Calendar Utils > generateCalendar > should generate the correct calendar for a month that starts on Friday and the first day of week is Monday 1`] = `
{
  "days": [
    [
      {
        "date": 27,
        "month": 10,
        "weekday": 1,
      },
      {
        "date": 28,
        "month": 10,
        "weekday": 2,
      },
      {
        "date": 29,
        "month": 10,
        "weekday": 3,
      },
      {
        "date": 30,
        "month": 10,
        "weekday": 4,
      },
      {
        "date": 1,
        "month": 11,
        "weekday": 5,
      },
      {
        "date": 2,
        "month": 11,
        "weekday": 6,
      },
      {
        "date": 3,
        "month": 11,
        "weekday": 0,
      },
    ],
    [
      {
        "date": 4,
        "month": 11,
        "weekday": 1,
      },
      {
        "date": 5,
        "month": 11,
        "weekday": 2,
      },
      {
        "date": 6,
        "month": 11,
        "weekday": 3,
      },
      {
        "date": 7,
        "month": 11,
        "weekday": 4,
      },
      {
        "date": 8,
        "month": 11,
        "weekday": 5,
      },
      {
        "date": 9,
        "month": 11,
        "weekday": 6,
      },
      {
        "date": 10,
        "month": 11,
        "weekday": 0,
      },
    ],
    [
      {
        "date": 11,
        "month": 11,
        "weekday": 1,
      },
      {
        "date": 12,
        "month": 11,
        "weekday": 2,
      },
      {
        "date": 13,
        "month": 11,
        "weekday": 3,
      },
      {
        "date": 14,
        "month": 11,
        "weekday": 4,
      },
      {
        "date": 15,
        "month": 11,
        "weekday": 5,
      },
      {
        "date": 16,
        "month": 11,
        "weekday": 6,
      },
      {
        "date": 17,
        "month": 11,
        "weekday": 0,
      },
    ],
    [
      {
        "date": 18,
        "month": 11,
        "weekday": 1,
      },
      {
        "date": 19,
        "month": 11,
        "weekday": 2,
      },
      {
        "date": 20,
        "month": 11,
        "weekday": 3,
      },
      {
        "date": 21,
        "month": 11,
        "weekday": 4,
      },
      {
        "date": 22,
        "month": 11,
        "weekday": 5,
      },
      {
        "date": 23,
        "month": 11,
        "weekday": 6,
      },
      {
        "date": 24,
        "month": 11,
        "weekday": 0,
      },
    ],
    [
      {
        "date": 25,
        "month": 11,
        "weekday": 1,
      },
      {
        "date": 26,
        "month": 11,
        "weekday": 2,
      },
      {
        "date": 27,
        "month": 11,
        "weekday": 3,
      },
      {
        "date": 28,
        "month": 11,
        "weekday": 4,
      },
      {
        "date": 29,
        "month": 11,
        "weekday": 5,
      },
      {
        "date": 30,
        "month": 11,
        "weekday": 6,
      },
      {
        "date": 31,
        "month": 11,
        "weekday": 0,
      },
    ],
  ],
  "firstDayOfWeek": 1,
}
`;

exports[`Calendar Utils > generateCalendar > should generate the correct calendar for a month that starts on Monday and the first day of week is Monday 1`] = `
{
  "days": [
    [
      {
        "date": 1,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 2,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 3,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 4,
        "month": 4,
        "weekday": 4,
      },
      {
        "date": 5,
        "month": 4,
        "weekday": 5,
      },
      {
        "date": 6,
        "month": 4,
        "weekday": 6,
      },
      {
        "date": 7,
        "month": 4,
        "weekday": 0,
      },
    ],
    [
      {
        "date": 8,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 9,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 10,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 11,
        "month": 4,
        "weekday": 4,
      },
      {
        "date": 12,
        "month": 4,
        "weekday": 5,
      },
      {
        "date": 13,
        "month": 4,
        "weekday": 6,
      },
      {
        "date": 14,
        "month": 4,
        "weekday": 0,
      },
    ],
    [
      {
        "date": 15,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 16,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 17,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 18,
        "month": 4,
        "weekday": 4,
      },
      {
        "date": 19,
        "month": 4,
        "weekday": 5,
      },
      {
        "date": 20,
        "month": 4,
        "weekday": 6,
      },
      {
        "date": 21,
        "month": 4,
        "weekday": 0,
      },
    ],
    [
      {
        "date": 22,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 23,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 24,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 25,
        "month": 4,
        "weekday": 4,
      },
      {
        "date": 26,
        "month": 4,
        "weekday": 5,
      },
      {
        "date": 27,
        "month": 4,
        "weekday": 6,
      },
      {
        "date": 28,
        "month": 4,
        "weekday": 0,
      },
    ],
    [
      {
        "date": 29,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 30,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 31,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 1,
        "month": 5,
        "weekday": 4,
      },
      {
        "date": 2,
        "month": 5,
        "weekday": 5,
      },
      {
        "date": 3,
        "month": 5,
        "weekday": 6,
      },
      {
        "date": 4,
        "month": 5,
        "weekday": 0,
      },
    ],
  ],
  "firstDayOfWeek": 1,
}
`;

exports[`Calendar Utils > generateCalendar > should generate the correct calendar for a month that starts on Monday and the first day of week is Saturday 1`] = `
{
  "days": [
    [
      {
        "date": 29,
        "month": 3,
        "weekday": 6,
      },
      {
        "date": 30,
        "month": 3,
        "weekday": 0,
      },
      {
        "date": 1,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 2,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 3,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 4,
        "month": 4,
        "weekday": 4,
      },
      {
        "date": 5,
        "month": 4,
        "weekday": 5,
      },
    ],
    [
      {
        "date": 6,
        "month": 4,
        "weekday": 6,
      },
      {
        "date": 7,
        "month": 4,
        "weekday": 0,
      },
      {
        "date": 8,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 9,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 10,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 11,
        "month": 4,
        "weekday": 4,
      },
      {
        "date": 12,
        "month": 4,
        "weekday": 5,
      },
    ],
    [
      {
        "date": 13,
        "month": 4,
        "weekday": 6,
      },
      {
        "date": 14,
        "month": 4,
        "weekday": 0,
      },
      {
        "date": 15,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 16,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 17,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 18,
        "month": 4,
        "weekday": 4,
      },
      {
        "date": 19,
        "month": 4,
        "weekday": 5,
      },
    ],
    [
      {
        "date": 20,
        "month": 4,
        "weekday": 6,
      },
      {
        "date": 21,
        "month": 4,
        "weekday": 0,
      },
      {
        "date": 22,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 23,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 24,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 25,
        "month": 4,
        "weekday": 4,
      },
      {
        "date": 26,
        "month": 4,
        "weekday": 5,
      },
    ],
    [
      {
        "date": 27,
        "month": 4,
        "weekday": 6,
      },
      {
        "date": 28,
        "month": 4,
        "weekday": 0,
      },
      {
        "date": 29,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 30,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 31,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 1,
        "month": 5,
        "weekday": 4,
      },
      {
        "date": 2,
        "month": 5,
        "weekday": 5,
      },
    ],
  ],
  "firstDayOfWeek": 6,
}
`;

exports[`Calendar Utils > generateCalendar > should generate the correct calendar for a month that starts on Monday and the first day of week is Sunday 1`] = `
{
  "days": [
    [
      {
        "date": 30,
        "month": 3,
        "weekday": 0,
      },
      {
        "date": 1,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 2,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 3,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 4,
        "month": 4,
        "weekday": 4,
      },
      {
        "date": 5,
        "month": 4,
        "weekday": 5,
      },
      {
        "date": 6,
        "month": 4,
        "weekday": 6,
      },
    ],
    [
      {
        "date": 7,
        "month": 4,
        "weekday": 0,
      },
      {
        "date": 8,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 9,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 10,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 11,
        "month": 4,
        "weekday": 4,
      },
      {
        "date": 12,
        "month": 4,
        "weekday": 5,
      },
      {
        "date": 13,
        "month": 4,
        "weekday": 6,
      },
    ],
    [
      {
        "date": 14,
        "month": 4,
        "weekday": 0,
      },
      {
        "date": 15,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 16,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 17,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 18,
        "month": 4,
        "weekday": 4,
      },
      {
        "date": 19,
        "month": 4,
        "weekday": 5,
      },
      {
        "date": 20,
        "month": 4,
        "weekday": 6,
      },
    ],
    [
      {
        "date": 21,
        "month": 4,
        "weekday": 0,
      },
      {
        "date": 22,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 23,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 24,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 25,
        "month": 4,
        "weekday": 4,
      },
      {
        "date": 26,
        "month": 4,
        "weekday": 5,
      },
      {
        "date": 27,
        "month": 4,
        "weekday": 6,
      },
    ],
    [
      {
        "date": 28,
        "month": 4,
        "weekday": 0,
      },
      {
        "date": 29,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 30,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 31,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 1,
        "month": 5,
        "weekday": 4,
      },
      {
        "date": 2,
        "month": 5,
        "weekday": 5,
      },
      {
        "date": 3,
        "month": 5,
        "weekday": 6,
      },
    ],
  ],
  "firstDayOfWeek": 0,
}
`;

exports[`Calendar Utils > generateCalendar > should generate the correct calendar for a month that starts on Saturday and the first day of week is Sunday 1`] = `
{
  "days": [
    [
      {
        "date": 26,
        "month": 2,
        "weekday": 0,
      },
      {
        "date": 27,
        "month": 2,
        "weekday": 1,
      },
      {
        "date": 28,
        "month": 2,
        "weekday": 2,
      },
      {
        "date": 29,
        "month": 2,
        "weekday": 3,
      },
      {
        "date": 30,
        "month": 2,
        "weekday": 4,
      },
      {
        "date": 31,
        "month": 2,
        "weekday": 5,
      },
      {
        "date": 1,
        "month": 3,
        "weekday": 6,
      },
    ],
    [
      {
        "date": 2,
        "month": 3,
        "weekday": 0,
      },
      {
        "date": 3,
        "month": 3,
        "weekday": 1,
      },
      {
        "date": 4,
        "month": 3,
        "weekday": 2,
      },
      {
        "date": 5,
        "month": 3,
        "weekday": 3,
      },
      {
        "date": 6,
        "month": 3,
        "weekday": 4,
      },
      {
        "date": 7,
        "month": 3,
        "weekday": 5,
      },
      {
        "date": 8,
        "month": 3,
        "weekday": 6,
      },
    ],
    [
      {
        "date": 9,
        "month": 3,
        "weekday": 0,
      },
      {
        "date": 10,
        "month": 3,
        "weekday": 1,
      },
      {
        "date": 11,
        "month": 3,
        "weekday": 2,
      },
      {
        "date": 12,
        "month": 3,
        "weekday": 3,
      },
      {
        "date": 13,
        "month": 3,
        "weekday": 4,
      },
      {
        "date": 14,
        "month": 3,
        "weekday": 5,
      },
      {
        "date": 15,
        "month": 3,
        "weekday": 6,
      },
    ],
    [
      {
        "date": 16,
        "month": 3,
        "weekday": 0,
      },
      {
        "date": 17,
        "month": 3,
        "weekday": 1,
      },
      {
        "date": 18,
        "month": 3,
        "weekday": 2,
      },
      {
        "date": 19,
        "month": 3,
        "weekday": 3,
      },
      {
        "date": 20,
        "month": 3,
        "weekday": 4,
      },
      {
        "date": 21,
        "month": 3,
        "weekday": 5,
      },
      {
        "date": 22,
        "month": 3,
        "weekday": 6,
      },
    ],
    [
      {
        "date": 23,
        "month": 3,
        "weekday": 0,
      },
      {
        "date": 24,
        "month": 3,
        "weekday": 1,
      },
      {
        "date": 25,
        "month": 3,
        "weekday": 2,
      },
      {
        "date": 26,
        "month": 3,
        "weekday": 3,
      },
      {
        "date": 27,
        "month": 3,
        "weekday": 4,
      },
      {
        "date": 28,
        "month": 3,
        "weekday": 5,
      },
      {
        "date": 29,
        "month": 3,
        "weekday": 6,
      },
    ],
    [
      {
        "date": 30,
        "month": 3,
        "weekday": 0,
      },
      {
        "date": 1,
        "month": 4,
        "weekday": 1,
      },
      {
        "date": 2,
        "month": 4,
        "weekday": 2,
      },
      {
        "date": 3,
        "month": 4,
        "weekday": 3,
      },
      {
        "date": 4,
        "month": 4,
        "weekday": 4,
      },
      {
        "date": 5,
        "month": 4,
        "weekday": 5,
      },
      {
        "date": 6,
        "month": 4,
        "weekday": 6,
      },
    ],
  ],
  "firstDayOfWeek": 0,
}
`;

exports[`Calendar Utils > generateCalendar > should generate the correct calendar for a month that starts on Thursday and the first day of week is Monday 1`] = `
{
  "days": [
    [
      {
        "date": 28,
        "month": 10,
        "weekday": 1,
      },
      {
        "date": 29,
        "month": 10,
        "weekday": 2,
      },
      {
        "date": 30,
        "month": 10,
        "weekday": 3,
      },
      {
        "date": 1,
        "month": 11,
        "weekday": 4,
      },
      {
        "date": 2,
        "month": 11,
        "weekday": 5,
      },
      {
        "date": 3,
        "month": 11,
        "weekday": 6,
      },
      {
        "date": 4,
        "month": 11,
        "weekday": 0,
      },
    ],
    [
      {
        "date": 5,
        "month": 11,
        "weekday": 1,
      },
      {
        "date": 6,
        "month": 11,
        "weekday": 2,
      },
      {
        "date": 7,
        "month": 11,
        "weekday": 3,
      },
      {
        "date": 8,
        "month": 11,
        "weekday": 4,
      },
      {
        "date": 9,
        "month": 11,
        "weekday": 5,
      },
      {
        "date": 10,
        "month": 11,
        "weekday": 6,
      },
      {
        "date": 11,
        "month": 11,
        "weekday": 0,
      },
    ],
    [
      {
        "date": 12,
        "month": 11,
        "weekday": 1,
      },
      {
        "date": 13,
        "month": 11,
        "weekday": 2,
      },
      {
        "date": 14,
        "month": 11,
        "weekday": 3,
      },
      {
        "date": 15,
        "month": 11,
        "weekday": 4,
      },
      {
        "date": 16,
        "month": 11,
        "weekday": 5,
      },
      {
        "date": 17,
        "month": 11,
        "weekday": 6,
      },
      {
        "date": 18,
        "month": 11,
        "weekday": 0,
      },
    ],
    [
      {
        "date": 19,
        "month": 11,
        "weekday": 1,
      },
      {
        "date": 20,
        "month": 11,
        "weekday": 2,
      },
      {
        "date": 21,
        "month": 11,
        "weekday": 3,
      },
      {
        "date": 22,
        "month": 11,
        "weekday": 4,
      },
      {
        "date": 23,
        "month": 11,
        "weekday": 5,
      },
      {
        "date": 24,
        "month": 11,
        "weekday": 6,
      },
      {
        "date": 25,
        "month": 11,
        "weekday": 0,
      },
    ],
    [
      {
        "date": 26,
        "month": 11,
        "weekday": 1,
      },
      {
        "date": 27,
        "month": 11,
        "weekday": 2,
      },
      {
        "date": 28,
        "month": 11,
        "weekday": 3,
      },
      {
        "date": 29,
        "month": 11,
        "weekday": 4,
      },
      {
        "date": 30,
        "month": 11,
        "weekday": 5,
      },
      {
        "date": 31,
        "month": 11,
        "weekday": 6,
      },
      {
        "date": 1,
        "month": 0,
        "weekday": 0,
      },
    ],
  ],
  "firstDayOfWeek": 1,
}
`;
